jscomp/ml/matching.ml

```diff
@@ -2918,7 +2918,9 @@ let partial_function loc () =
   let fname = 
     Filename.basename fname
   in   
-  Lprim(Praise Raise_regular, [Lprim(Pmakeblock(Blk_extension),
+  Lprim(Praise Raise_regular, [Lprim(Pmakeblock(Blk_extension {
+    is_exception = true;
```

Comment:
How do you know that there's no other place in the compiler where one could have forgotten to do this?
Are there enough tests?

Reply:
E.g. `let e = E(10)`

Reply:
Are you talking about adding the correct `is_exception` value? There is compiler complaining in every place where it's needed + there's `record_extension_test.js` and multiple other tests showing that output is correct.